jfr/src/main/java/io/netty/jfr/JfrChannelHandler.java

```diff
@@ -0,0 +1,383 @@
+            channelUnregisteredEvent.commit();
+        }
+    }
+
+    @Override
+    public void channelActive(ChannelHandlerContext ctx) {
+        if (wrapInChannelEvent(channelActiveEvent, ctx, ctx::fireChannelActive)) {
+            channelActiveEvent.commit();
+        }
+    }
+
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) {
+        if (wrapInChannelEvent(channelInactiveEvent, ctx, ctx::fireChannelInactive)) {
+            channelInactiveEvent.commit();
+        }
+    }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+        if (wrapInChannelEvent(exceptionCaughtEvent, ctx, () -> ctx.fireExceptionCaught(cause), () -> {
+        }, () -> exceptionCaughtEvent.setCause(cause))) {
+            exceptionCaughtEvent.commit();
+        }
+    }
+
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
+        if (wrapInChannelEvent(userEventTriggeredEvent, ctx, () -> ctx.fireUserEventTriggered(evt),
+                               () -> userEventTriggeredEvent.setUserEvent(evt), () -> {
+                })) {
+            userEventTriggeredEvent.commit();
+        }
+    }
+
+    @Override
+    public void bind(ChannelHandlerContext ctx, SocketAddress localAddress, ChannelPromise promise) {
+        if (wrapInChannelEvent(bindEvent, ctx, () -> ctx.bind(localAddress, promise), () -> {
+        }, () -> bindEvent.setLocalAddress(localAddress))) {
+            bindEvent.commit();
+        }
+    }
+
+    @Override
+    public void connect(
+            ChannelHandlerContext ctx,
+            SocketAddress remoteAddress, SocketAddress localAddress, ChannelPromise promise) {
+        if (wrapInChannelEvent(connectEvent, ctx, () -> ctx.connect(remoteAddress, localAddress, promise), () -> {
+                               },
+                               () -> {
+                                   connectEvent.setRemoteAddress(remoteAddress);
+                                   connectEvent.setLocalAddress(localAddress);
+                               })) {
+            connectEvent.commit();
+        }
+    }
+
+    @Override
+    public void disconnect(ChannelHandlerContext ctx, ChannelPromise promise) {
+        if (wrapInChannelEvent(disconnectEvent, ctx, () -> ctx.disconnect(promise))) {
+            disconnectEvent.commit();
+        }
+    }
+
+    @Override
+    public void close(ChannelHandlerContext ctx, ChannelPromise promise) {
+        if (wrapInChannelEvent(closeEvent, ctx, () -> ctx.close(promise))) {
+            closeEvent.commit();
+        }
+    }
+
+    @Override
+    public void deregister(ChannelHandlerContext ctx, ChannelPromise promise) {
+        if (wrapInChannelEvent(deregisterEvent, ctx, () -> ctx.deregister(promise))) {
+            deregisterEvent.commit();
+        }
+    }
+
+// NOTE: Occurs so often, currently removed for performance reasons, will try to optimize and add.
```

Comment:
is this still an issue ?

Reply:
I benchmarked a bit, and it is still an issue that I need to solve, or remove the commented code.

Reply:
Imho it would be kind of odd to not have all events produced

Reply:
I agree. I believe when I looked at the io.netty.handler.logging.LoggingHandler, it does not override read() (so I was thinking I could skip that one), but it does include channelReadComplete.
I just have to make sure the channelReadComplete doesn't cause to much overhead even when it's disabled.

Reply:
@klaraward hmm... LoggingHandler does override read :)

Reply:
io.netty.channel.LoggingHandler does, io.netty.handler.logging.LoggingHandler does not (I think this is the one I originally looked at)